torch/csrc/jit/tensorexpr/tensorexpr_tutorical.py

```diff
@@ -0,0 +1,219 @@
+# Tensor Expression: python API
+#
+# This tutorial explains the python API of TE that includes: 1) TE IR, 2) TE
+# loop transformations, and 3) TE codegen.
+
+import torch
+import torch._C._te as te
+from numpy import median
+import timeit
+
+LLVM_ENABLED = torch._C._llvm_enabled()
+
+# 1. End-to-End example: addition of two tensors
+
+# We first create two BufHandles in TE which correspond to the two input
+# tensors.
+
+# The data (physical memory) of a tensor is represented as a buffer
+# in TE, which has dimensions, a data type and a name. BufHandle is a pointer
+# to a buffer that defines load/store operations on it. Here, A points to a
+# buffer with dimentions [32], 'float' data type and name 'A'; B points to a
+# buffer with dimentions [32], 'float' data type and name 'B'.
+A = te.BufHandle("A", [32], torch.float32)
+B = te.BufHandle("B", [32], torch.float32)
+
+# We then define the addition operation of two BufHandles: 'add'. It takes the
+# axis argument 'i' and returns the value of the ouput buffer on 'i', i.e.,
+# A[i] + B[i]. A[i], B[i] correspond to 'A.load([i])', 'B.load([i])' in TE
+# respectively.
+def add(i):
+    return A.load([i]) + B.load([i])
+
+# Next, we define the output tensor using TE 'Compute' - it specifies the name,
+# dimensions of the output tensor, and how to compute its values, i.e., the
+# 'add' operation we constructed above.
+C = te.Compute("C", [32], add)
+
+# At this point, we have constructed the IR stmt that computes the tensor C.
+# Use the following code to print its stmt.
+print("The stmt to compute tensor C:")
+print(C.stmt())
+
+# The stmt to compute tensor C:
+# for (int v = 0; v < 32; v++) {
+#   C[v] = (A[v]) + (B[v]);
+# }
+
+# Lastly, we construct a loopnest for C to prepare for codegen. Namely, we
+# transform the stmt to 1) expand reduce operations, and 2) flatten buffer
+# dimemsions. In this example, there are no reduces or multi-dimentional
```

Comment:
We haven't talked about reductions at this point and thus mentioning "expand reduce operations" might be confusing for the readers. Do we even need to tell what exactly happens in prepare for codegen?

Also, in this particular example I think we can skip these steps entirely as they do nothing - this would make an example shorter and easier to follow.